homework/index.js

```diff
@@ -1,54 +1,106 @@
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
+  const repositorySection = document.querySelector('.repoContainer');
+  const contributorsSection = document.querySelector('.contributorsContainer');
 
+  function getFetch(url) {
+    fetch(url)
+      .then(response => {
+        const contentType = response.headers.get('content-type');
+        if (!contentType || !contentType.includes('json') || !response.ok) {
+          const errorMessage = new Error("didn't get JSON");
+          return Promise.reject(errorMessage);
+        }
+        return response.json();
+      })
+      .then(data => { (repositorySection.innerHTML === '') ? createOptionsOfSelect(data) : createContributors(data) })
+      .catch(errorMessage => (repositorySection.innerHTML === '') ? createErrorDiv(errorMessage, repositorySection) : createErrorDiv(errorMessage, contributorsSection));
+  }
   function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
+    const element = document.createElement(name);
+    parent.appendChild(element);
     Object.entries(options).forEach(([key, value]) => {
       if (key === 'text') {
-        elem.textContent = value;
+        element.innerHTML = value;
       } else {
-        elem.setAttribute(key, value);
+        element.setAttribute(key, value);
       }
     });
-    return elem;
+    return element;
+  }
+  function createErrorDiv(errorMessage, sectionName) {
+    createAndAppend('div', sectionName, {
+      text: errorMessage,
+      class: 'alert-error',
+    });
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+
+  function createHeader() {
+    const header = document.querySelector('header');
+    createAndAppend('h2', header, { text: 'Hack Your Future Repositories', id: 'hyf-header' });
+    createAndAppend('select', header);
+    const select = document.querySelector('select');
+    createAndAppend('option', select, { text: "Hack Your Future Repositories", value: 0 });
   }
+  function createOptionsOfSelect(data) {
+    const select = document.querySelector('select');
+    select.innerHTML = "";
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
-        });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+    const sortedRepositories = data.sort((a, b) => a.name.localeCompare(b.name, 'en', { numeric: true }, { sensitivity: 'base' }));
+    sortedRepositories.forEach((data, index, arr) => {
+      createAndAppend('option', select, {
+        text: `${data.name}`,
+        value: `${index}`
+      });
+    });
+    onChangeSelect(0, data);
```

Comment:
This function name doesn't tell me what's going to happen. It actually only confused me 😳 
```suggestion
    updateRepositoryDetails(data);
```